woodwork/logical_types.py

```diff
@@ -199,8 +199,7 @@ def transform(self, series):
         """Converts the series data to a formatted datetime. Datetime format will be inferred if datetime_format is None."""
         new_dtype = self._get_valid_dtype(type(series))
         if new_dtype != str(series.dtype):
```

Comment:
Noticed that the fact that the `_infer_datetime_format` call happens after this check means that the datetime format is not always saved. Opened an issue #1138